lxd/container.go

```diff
@@ -684,8 +684,8 @@ func instanceCreateAsCopy(s *state.State, args db.InstanceArgs, sourceInst Insta
 				return nil, err
 			}
 
-			// Restore snapshot creation date.
-			err = s.Cluster.ContainerCreationUpdate(snapInst.ID(), snap.CreationDate())
+			// Set snapshot creation date to that of the source snapshot.
+			err = s.Cluster.InstanceSnapshotCreationUpdate(snapInst.ID(), srcSnap.CreationDate())
```

Comment:
@stgraber this line also fixes a bug in updating the new snapshot's creation date time to the source snapshot, because it was using ContainerCreationUpdate with an instance snapshot ID. This actually might have meant that unrelated instance's creation times would be updated in error in the current implementation.